src/python/pants/backend/python/lint/bandit/rules.py

```diff
@@ -26,21 +26,22 @@
 from pants.engine.process import FallibleProcessResult, Process
 from pants.engine.rules import SubsystemRule, named_rule
 from pants.engine.selectors import Get, MultiGet
+from pants.engine.target import FieldSetWithOrigin
 from pants.engine.unions import UnionRule
 from pants.option.global_options import GlobMatchErrorBehavior
 from pants.python.python_setup import PythonSetup
 from pants.util.strutil import pluralize
 
 
 @dataclass(frozen=True)
-class BanditFieldSet(LinterFieldSet):
+class BanditFieldSet(FieldSetWithOrigin):
```

Comment:
Nice, this is simpler. 

Normally, we want union members to subclass the upstream type, but a `LinterFieldSet` was never part of a union, so it ends up being more complex than necessary.